tawazi/decorators.py

```diff
@@ -145,7 +144,7 @@ def dag(
     # wrapper used to support parametrized and non parametrized decorators
     def intermediate_wrapper(_func: Callable[P, RVDAG]) -> DAG[P, RVDAG]:
         # 0. Protect against multiple threads declaring many DAGs at the same time
-        with exec_nodes_lock:
+        with node.exec_nodes_lock:
```

Comment:
can't we import the lock from the module now that there are no more circular imports ?

Reply:
hmm I don't really know then